src/Symfony/Component/Mailer/Transport.php

```diff
@@ -64,8 +64,8 @@ private static function createTransport(string $dsn, EventDispatcherInterface $d
             throw new InvalidArgumentException(sprintf('The "%s" mailer DSN is invalid.', $dsn));
         }
 
-        if (!isset($parsedDsn['host'])) {
-            throw new InvalidArgumentException(sprintf('The "%s" mailer DSN must contain a mailer name.', $dsn));
+        if (!isset($parsedDsn['scheme'], $parsedDsn['host'])) {
```

Comment:
I suggest making checks separately, so that we can have dedicated error message instead of squashing the 2 error cases together.

Reply:
I thought about that at first, but I've had a hard time coming up with a test case that contained a scheme but no host. Turms out, `file:///` is the only scheme where php allows omitting the host. I've split the check for now, but I'm uncertain if we really gain anything from that.